test/end-to-end/incident-query-filter.test.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+var utils = require('./e2e-tools');
+var chai = require('chai');
+var chaiAsPromised = require('chai-as-promised');
+chai.use(chaiAsPromised);
+var expect = chai.expect;
+var promise = protractor.promise;
+var reports = require('./fixtures/smsgh_reports.json');
+var incidents = require('./fixtures/incidents.json');
+
+// Allow chai to wait for promises on the right-hand-side
+chaiAsPromised.transformAsserterArgs = function(args) {
+  return promise.all(args);
+};
+
+describe.only('Incident query filter', function() {
+  before(utils.initDb);
+  after(utils.disconnectDropDb);
+
+  beforeEach(utils.resetDb);
+  beforeEach(utils.initAdmin.bind({}, 'asdfasdf'));
+  beforeEach(utils.toggleFetching.bind({}, 'Off'));
+  beforeEach(utils.addSource.bind({}, 'SMS GH', { nickname: 'foo', keywords: 'test' }));
+  beforeEach(utils.toggleFetching.bind({}, 'On'));
+
+  afterEach(utils.deleteSource.bind({}, 'SMS GH', 'foo'));
+  afterEach(utils.toggleFetching.bind({}, 'Off'));
+  afterEach(utils.resetBrowser);
+
+  var reqParams1 = reports[0];
+  var reqParams2 = reports[1];
+  var reqParams3 = reports[2];
+  var reqParams4 = reports[3];
```

Comment:
Can we just find & replace these to clean this up? It'll make codeclimate happy too :)